generic.go

```diff
@@ -42,6 +47,86 @@ func (api API) create(path string, input interface{}, output interface{}) error
 	return wrapErrors(res, body, errs)
 }
 
+func (api API) prepareMultiPartRequest(path string, filePath string, input interface{}) (*http.Request, error) {
```

Comment:
Let's split it on several functions. File manipulations we can move out or sure.